script/script.js

```diff
@@ -136,44 +154,177 @@ const form = {
             view: "text",
             label: "Rating",
             name: "rating",
-            invalidMessage: "Cannot be empty or 0"
+            invalidMessage: "Field cannot be empty or 0."
         },
         {
             view: "text",
             label: "Votes",
             name: "votes",
-            invalidMessage: "Cannot be less than 100000"
+            invalidMessage: "Cannot be more than 100000"
         },
         formButtons,
         {}
     ],
     rules: {
         title: webix.rules.isNotEmpty,
         year: function (value) {
-            return value >= 1970 && value <= new Date().getFullYear();
+            return value >= 1970 && value <= currentYear && webix.rules.isNumber(value);
         },
         votes: function (value) {
-            return value < 100000;
+            return value <= 100000 && webix.rules.isNumber(value);
         },
         rating: function (value) {
             return webix.rules.isNotEmpty(value) && value !== "0";
         }
     }
 };
+
+// adding item's data to form fields
+const valuesToForm = function (id) {
+    const selectedItemValues = $$("datatable").getItem(id);
+    $$("form").setValues(selectedItemValues);
+};
+
+const datatable = {
+    view: "datatable",
+    id: "datatable",
+    scroll: "y",
+    select: true,
+    url: "../data/data.js",
+    hover: "hover-style",
+    sort: "multi",
+    columns: datatableColumns,
+    onClick: {
+        "wxi-trash": function (e, id) {
+            this.remove(id);
+            if ($$("form").getValues()) {
```

Comment:
here you clear the form anyway after deleting, but this makes sense only when the form contains the deleted record (you can compare ids)